server/src/test/java/org/elasticsearch/action/admin/cluster/reroute/ClusterRerouteResponseTests.java

```diff
@@ -93,28 +95,28 @@ public void testToXContent() throws IOException {
                     "      \"templates\" : { },\n" +
                     "      \"indices\" : {\n" +
                     "        \"index\" : {\n" +
+                    "          \"version\" : 1,\n" +
+                    "          \"mapping_version\" : 1,\n" +
+                    "          \"settings_version\" : 1,\n" +
+                    "          \"aliases_version\" : 1,\n" +
+                    "          \"routing_num_shards\" : 1,\n" +
                     "          \"state\" : \"open\",\n" +
                     "          \"settings\" : {\n" +
-                    "            \"index\" : {\n" +
-                    "              \"shard\" : {\n" +
-                    "                \"check_on_startup\" : \"true\"\n" +
-                    "              },\n" +
-                    "              \"number_of_shards\" : \"1\",\n" +
-                    "              \"number_of_replicas\" : \"0\",\n" +
-                    "              \"version\" : {\n" +
-                    "                \"created\" : \"" + Version.CURRENT.id + "\"\n" +
-                    "              },\n" +
-                    "              \"max_script_fields\" : \"10\"\n" +
-                    "            }\n" +
-                    "          },\n" +
-                    "          \"mappings\" : { },\n" +
-                    "          \"aliases\" : [ ],\n" +
-                    "          \"primary_terms\" : {\n" +
-                    "            \"0\" : 0\n" +
+                    "            \"index.max_script_fields\" : \"10\",\n" +
```

Comment:
MetaData.toXContent could have `flat_settings` defaulting to `false` if the context is API. That means that the representation would stay the same here.